mantidimaging/core/operations/rotate_stack/rotate_stack.py

```diff
@@ -96,15 +95,94 @@ def _update_angle(angle, dropdown):
+    rotated by.
+
+    param: value: angle of rotation
+    return: angle: cardinal angle of rotation
+    """
+    angle = ((value % 360 + 45) // 90 * 90 - (90 if (value % 360) % 90 == 45 else 0))
+    return angle % 360
+
+
+def _compute_rotation(data, angle, progress) -> np.ndarray:
+    """
+    Handle rotation of image slice by angle. This includes rotation of aspect
+    ratio and in place rotation depending on angle.
+
+    param array_index: index of array in array_list
+    param data: current image data array
+    param angle: angle of rotation
+    return: new_data: rotated image data array
+    """
+    angle += 360 if angle < 0 else 0
+    if angle == _get_cardinal_angle(angle):
+        data.shared_array = _cardinal_rotation_per_slice(data, angle, progress)
+    else:
+        data.shared_array = _cardinal_rotation_per_slice(data, _get_cardinal_angle(angle), progress)
+        data.shared_array = _inplace_rotation(data.data.shape[0], data, angle, _get_cardinal_angle(angle), progress)
+    return data
+
+
+def _cardinal_rotation_per_slice(data: ImageStack, angle: float, progress) -> 'pu.SharedArray':
+    """
+    Perform a cardinal rotation of the image stack by angle if angle is cardinal and additionally
+    in-place rotation if general angle.
+
+    param: data: current image data array
+    param: angle: angle of rotation
+    param: progress: progress bar
+    return: new_data: rotated image data array
+    """
+    z_axis, y_axis, x_axis = data.data.shape
+    rotated_shape: tuple = (z_axis, x_axis, y_axis)
 
+    if angle == 0 or angle == 180:
+        new_data = _inplace_rotation(z_axis, data, angle, 0, progress)
+    else:
+        new_data = pu.create_array(rotated_shape, data.data.dtype)
+        ps.run_compute_func(_compute_cardinal_rotation_per_slice, z_axis, [data.shared_array, new_data],
+                            {"angle": angle}, progress)
+    return new_data
 
-def _execute(images: ImageStack, angle: float, progress: Progress):
-    progress = Progress.ensure_instance(progress, task_name='Rotate Stack')
 
-    with progress:
-        f = ps.create_partial(_rotate_image_inplace, ps.inplace1, angle=angle)
-        ps.execute(f, [images.shared_array], images.data.shape[0], progress, msg=f"Rotating by {angle} degrees")
+def _inplace_rotation(array_index_z: int, data, angle: float, angle_offset: int, progress) -> 'pu.SharedArray':
+    """
+    In-place rotation of image slices by angle.
+
+    param: array_index_z: index of array in array_list
+    param: data: current image data array
+    param: angle: angle of rotation
+    param: angle_offset: angle offset
+    param: progress: progress bar
+    return: new_data: rotated image data array
+    """
+    new_data = pu.create_array(data.data.shape, data.data.dtype)
+    ps.run_compute_func(_compute_rotation_per_slice_inplace, array_index_z, [data.shared_array, new_data],
+                        {"angle": angle - angle_offset}, progress)
```

Comment:
We should not need to create a new array here. Just need to pass in the existing data, and have the compute function write into the same function.